tests/src/test/scala/com/nvidia/spark/rapids/AdaptiveQueryExecSuite.scala

```diff
@@ -254,13 +254,18 @@ class AdaptiveQueryExecSuite
         _.isInstanceOf[GpuDataWritingCommandExec])
       assert(writeCommand.isDefined)
 
-      // the read should be an adaptive plan
-      val adaptiveSparkPlanExec = TestUtils.findOperator(writeCommand.get,
-        _.isInstanceOf[AdaptiveSparkPlanExec])
-        .get.asInstanceOf[AdaptiveSparkPlanExec]
+      if (cmpSparkVersion(3,4,0) < 0) {
+        // the read should be an adaptive plan prior to Spark 3.4.0
+        val adaptiveSparkPlanExec = TestUtils.findOperator(writeCommand.get,
+          _.isInstanceOf[AdaptiveSparkPlanExec])
+          .get.asInstanceOf[AdaptiveSparkPlanExec]
 
-      // assert that at least part of the adaptive plan ran on GPU
-      assert(TestUtils.findOperator(adaptiveSparkPlanExec, _.isInstanceOf[GpuExec]).isDefined)
+        // assert that at least part of the adaptive plan ran on GPU
+        assert(TestUtils.findOperator(adaptiveSparkPlanExec, _.isInstanceOf[GpuExec]).isDefined)
+      } else {
+        assert(TestUtils.findOperator(writeCommand.get,
+          _.isInstanceOf[GpuFileSourceScanExec]).isDefined)
```

Comment:
Shall we need to consider if the v2 data source becomes the default one, then the reader will be a `GpuBatchScanExec` ?

Reply:
This is a good consideration. Could you please create an issue to track this, @firestarman ?